packages/test/generatedCode/models/microsoft/graph/dateTimeTimeZone.ts

```diff
@@ -1,78 +1,9 @@
-    private _timeZone?: string | undefined;
-    /**
-     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
-     * @returns a Record<string, unknown>
-     */
-    public get additionalData() {
-        return this._additionalData;
-    };
-    /**
-     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
-     * @param value Value to set for the AdditionalData property.
-     */
-    public set additionalData(value: Record<string, unknown>) {
-        this._additionalData = value;
-    };
-    /**
-     * Instantiates a new dateTimeTimeZone and sets the default values.
-     */
-    public constructor() {
-        this._additionalData = {};
-    };
-    /**
-     * Gets the dateTime property value. A single point of time in a combined date and time representation ({date}T{time}; for example, 2017-08-29T04:00:00.0000000).
-     * @returns a string
-     */
-    public get dateTime() {
-        return this._dateTime;
-    };
-    /**
-     * Sets the dateTime property value. A single point of time in a combined date and time representation ({date}T{time}; for example, 2017-08-29T04:00:00.0000000).
-     * @param value Value to set for the dateTime property.
-     */
-    public set dateTime(value: string | undefined) {
-        this._dateTime = value;
-    };
-    /**
-     * The deserialization information for the current model
-     * @returns a Record<string, (node: ParseNode) => void>
-     */
-    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
-        return {
-            "dateTime": n => { this.dateTime = n.getStringValue(); },
-            "timeZone": n => { this.timeZone = n.getStringValue(); },
-        };
-    };
-    /**
-     * Serializes information the current object
-     * @param writer Serialization writer to use to serialize this model
-     */
-    public serialize(writer: SerializationWriter) : void {
-        if(!writer) throw new Error("writer cannot be undefined");
-        writer.writeStringValue("dateTime", this.dateTime);
-        writer.writeStringValue("timeZone", this.timeZone);
-        writer.writeAdditionalData(this.additionalData);
-    };
-    /**
-     * Gets the timeZone property value. Represents a time zone, for example, 'Pacific Standard Time'. See below for more possible values.
-     * @returns a string
-     */
-    public get timeZone() {
-        return this._timeZone;
-    };
-    /**
-     * Sets the timeZone property value. Represents a time zone, for example, 'Pacific Standard Time'. See below for more possible values.
-     * @param value Value to set for the timeZone property.
-     */
-    public set timeZone(value: string | undefined) {
-        this._timeZone = value;
-    };
+export interface DateTimeTimeZone{
```

Comment:
the interface should inherit from IParsable and IAdditionalPropertyHolder

Reply:
IParsable and IAdditionalPropertyHolder has required parameters which should be optional in the the model interface. 

In that case, an example as follows throws the errors that `additionalData` or `getFieldDeserializer` is not mentioned. 
```
const message = {
 subject: ""
}
```

The current implementation avoids that. 

The other option is to use `implements  Partial<IParsable> and Partial<AddnHolder>.

Reply:
I think the second option works well. What do you think?

Reply:
Yeah that's a good suggestion. An alternative would be to have the request body parameters and the return types to be intersection types: ModelInterface & Parsable & AdditionalPropertyHolder, so people know they also get those extra properties and can provide additional properties.

Reply:
Resolved